### PR TITLE
Fix ESLint plugin load error

### LINKED ISSUE #546: Error: Failed to load plugin '*' declared in 'BaseConfig...
With 0.40.1, xo --help works. But in my project folder, xo gives:
`Error: Failed to load plugin 'ava' declared in 'BaseConfig » /usr/lib/node_modules/xo/config/plugins.js': Cannot find module 'eslint-plugin-ava'`

npm list -g --depth 1 gives (for xo):
```
└─┬ xo@0.40.1
  ├── @eslint/eslintrc@0.4.1
  ├── @typescript-eslint/eslint-plugin@4.23.0
  ├── @typescript-eslint/parser@4.23.0
  ├── arrify@2.0.1
  ├── cosmiconfig@7.0.0
  ├── debug@4.3.1
  ├── define-lazy-prop@2.0.0
  ├── eslint-config-prettier@8.3.0
  ├── eslint-config-xo-typescript@0.41.1
  ├── eslint-config-xo@0.36.0
  ├── eslint-formatter-pretty@4.0.0
  ├── eslint-import-resolver-webpack@0.13.0
  ├── eslint-plugin-ava@12.0.0
  ├── eslint-plugin-eslint-comments@3.2.0
  ├── eslint-plugin-import@2.22.1
  ├── eslint-plugin-no-use-extend-native@0.5.0
  ├── eslint-plugin-node@11.1.0
  ├── eslint-plugin-prettier@3.4.0
  ├── eslint-plugin-promise@5.1.0
  ├── eslint-plugin-unicorn@32.0.1
  ├── eslint@7.26.0
  ├── find-cache-dir@3.3.1
  ├── find-up@5.0.0
  ├── fs-extra@10.0.0
  ├── get-stdin@8.0.0
  ├── globby@9.2.0
  ├── has-flag@4.0.0
  ├── imurmurhash@0.1.4
  ├── is-path-inside@3.0.3
  ├── json-stable-stringify-without-jsonify@1.0.1
  ├── json5@2.2.0
  ├── lodash@4.17.21
  ├── meow@9.0.0
  ├── micromatch@4.0.4
  ├── open-editor@3.0.0
  ├── p-filter@2.1.0
  ├── p-map@4.0.0
  ├── p-reduce@2.1.0
  ├── path-exists@4.0.0
  ├── prettier@2.3.0
  ├── resolve-cwd@3.0.0
  ├── semver@7.3.5
  ├── slash@3.0.0
  ├── to-absolute-glob@2.0.2
  ├── typescript@4.2.4
  └── update-notifier@5.1.0
```

So eslint-plugin-ava is there, but something isn't working.

```
Require stack:
- <...>/__placeholder__.js
Referenced from: /usr/lib/node_modules/xo/config/plugins.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:815:15)
    at Function.resolve (internal/modules/cjs/helpers.js:80:19)
    at Object.resolve (/usr/lib/node_modules/xo/node_modules/@eslint/eslintrc/lib/shared/relative-module-resolver.js:28:50)
    at ConfigArrayFactory._loadPlugin (/usr/lib/node_modules/xo/node_modules/@eslint/eslintrc/lib/config-array-factory.js:1017:39)
    at ConfigArrayFactory._loadExtendedPluginConfig (/usr/lib/node_modules/xo/node_modules/@eslint/eslintrc/lib/config-array-factory.js:837:29)
    at ConfigArrayFactory._loadExtends (/usr/lib/node_modules/xo/node_modules/@eslint/eslintrc/lib/config-array-factory.js:779:29)
    at ConfigArrayFactory._normalizeObjectConfigDataBody (/usr/lib/node_modules/xo/node_modules/@eslint/eslintrc/lib/config-array-factory.js:720:25)
    at _normalizeObjectConfigDataBody.next (<anonymous>)
    at ConfigArrayFactory._normalizeObjectConfigData (/usr/lib/node_modules/xo/node_modules/@eslint/eslintrc/lib/config-array-factory.js:665:20)
    at _normalizeObjectConfigData.next (<anonymous>)
```

_Originally posted by @AndrewMSHowe in https://github.com/xojs/xo/issues/543#issuecomment-838435953_

### PR DESCRIPTION
**Pull request by fisker**
Fixes #546

I think the only problem is,  user can override plugins before, but now can't, ESLint will always load the plugins `xo` installed.

I saw [a comment](https://github.com/xojs/xo/issues/546#issuecomment-863997813) from #546, but I'm not sure what the problem is.

### CODE REVIEW

File: lib/options-manager.js [hunk 1]
```diff
@@ -289,6 +290,9 @@ const buildESLintConfig = options => config => {
 		};
 	}
 
+	// Resolve plugins from XO root directory
+	config.resolvePluginsRelativeTo = MODULE_DIRECTORY;
```
Comment (sindresorhus) [inline lib/options-manager.js:294]: Doesn't this make it impossible for users to add ESLint plugins themselves?

Comment (fisker) [reply lib/options-manager.js:294]: I tested in `eslint-plugin-unicorn`, it can load `eslint-plugin-eslint-plugin`.

**Comment by sindresorhus** [comment]
> I saw a comment from #546, but I'm not sure what the problem is.

@Richienb Would you be able to comment on this? And review the PR.

**Comment by Richienb** [comment]
Does this also fix https://github.com/xojs/xo/issues/555?

**Comment by fisker** [comment]
I thought I found a better solution, but it didn't work.

I've [opened an issue in ESLint](https://github.com/eslint/eslint/issues/15079), if it get fixed, maybe we can pass plugins that `xo` installed, so ESLint won't try to load them.

**Comment by fregante** [comment]
The issue you opened upstream has been fixed 🚀

**Comment by fisker** [comment]
In that case this can be fixed by load plugins ourselves. I don't have time to work on this. It will be great if someone want give this a shot. BTW, ESLint have a new `FlatESLint` haven't play with it yet.

**Comment by fregante** [comment]
I'm going to close this since it's old. The flat config will eventually fix this issue anyway:


- #702

**Pull request closed without merging**